lib/puppet/util/windows/adsi.rb

```diff
@@ -464,12 +464,7 @@ def remove_member_sids(*sids)
     end
 
     def members
-      # WIN32OLE objects aren't enumerable, so no map
-      members = []
-      # Setting WIN32OLE.codepage in the microsoft_windows feature ensures
-      # values are returned as UTF-8
-      native_group.Members.each {|m| members << m.Name}
```

Comment:
Another plausible way of trying this is:

```ruby
native_group.Members.each {|m| members << m.AdsPath.gsub('WinNT://', '').gsub('/', '\\')}
```

While that's simpler than calling additional Windows APIs / populating additional objects, there is some useful logic inside of the Principal object already that creates a `domain_account` member that is in a form that Puppet can use - see https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/principal.rb#L16-L27